poh/src/poh_service.rs

```diff
@@ -130,8 +130,10 @@ impl PohService {
                     // PoH service runs in a tight loop, generating hashes as fast as possible.
                     // Let's dedicate one of the CPU cores to this thread so that it can gain
                     // from cache performance.
-                    if let Some(cores) = core_affinity::get_core_ids() {
-                        core_affinity::set_for_current(cores[pinned_cpu_core]);
+                    if let Ok(cpu_count) = agave_cpu_utils::cpu_count() {
```

Comment:
I think that the previous behavior (panic) is better than silently not setting
the affinity